src/main/java/org/zalando/nakadi/controller/SubscriptionController.java

```diff
@@ -143,4 +148,35 @@ private Subscription getExistingSubscription(final SubscriptionBase subscription
                 subscriptionBase.getConsumerGroup());
     }
 
+    private final class EventTypesHolder {
```

Comment:
@adyach This class is just two lists. And one of these lists you don't really need in my opinion, because you can always get it with something like:
```java
final List<EventType> existingEventTypes = getExistingEventTypes(subscriptionBase);
final List<String> missingEventTypes = subscriptionBase.getEventTypes()
        .stream()
        .filter(et -> existingEventTypes.stream().anyMatch(existingET -> existingET.getName().equals(et)))
        .collect(Collectors.toList());
```